lib/assessment.js

```diff
@@ -8,15 +8,45 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-var min = function min() {
 
+// iterate over the array, use ternary operator to compare currentValue
+// to previousValue and assign the min to whichever satisfies
+// the condition
+var min = function min(previousValue, currentValue, index, array) {
+  // set the intial result
+  var result = previousValue;
+  for (var i = 0; i < array.length; i++) {
+    // set the updated result
+    result = currentValue < previousValue ? currentValue : previousValue;
+  }
+  // return the value or it won't pass grunt test
+  return result;
```

Comment:
For a beginner programmer, if a function works *and* passes a grunt test, I don't see a reason for deduction points for wastefulness.